Qphix devel

plabus opened the pull request:
Implemented full solver interface for QPhiX CG for Wilson Fermions.

Solver converges, but real residuum is large. Possible issues:

1. All parameters set correctly? (Wilson mass)
2. Is BLAS linear algebra correct in preparing/reconstruction?
3. Is the rescaling tmlQCD <---> QPhiX correct?
4. Do we need additional Dslash applications, b/c we solving normal equations?

Commit:
Add MG4QCD to Inverters

Commit:
Add MG4QCD to Inverters

Commit:
add linalg-functions and hmc_control struct

Commit:
added MG4QCD to monomials: DET, DETRATIO and CLOVERDET

Commit:
Avoid fprintf if solver is not BICGSTAB

sbacchio committed:
Some parameters added to MG, solve squared sobstited by built in function (working with new version of MG4QCD, 1604)

sbacchio committed:
Merge branch 'MG4QCD' of https://github.com/Finkenrath/tmLQCD

sbacchio committed:
Deleted unecessary file generated by autoreconf

Commit:
Merge branch 'master' of ssh://github.com/Finkenrath/tmLQCD into MG4QCD

Commit:
MG4QCD interface updated: working with version 1605 of the library. Now MG is fully integreted in the HMC procedure and it has been tested.

Commit:
Fixed rel_prec for MG interface

sbacchio committed:
Cleaned MG interface and done TODO

Commit:
Cleaned MG interface and done TODO

sbacchio committed:
Merge branch 'MG4QCD' of ssh://github.com/Finkenrath/tmLQCD into MG4QCD

sbacchio committed:
Fixed some freeing issue

sbacchio committed:
Merge branch 'MG4QCD' of https://github.com/Finkenrath/tmLQCD into MG4QCD

sbacchio committed:
Fixed use of relative precision in MG interface

Commit:
Merge remote-tracking branch 'finkenrath/MG4QCD' into merge-MG4QCD

Conflicts:
	invert.c
	invert_clover_eo.c
	read_input.l
	solver/solver_types.h
	update_gauge.c

Commit:
mg4qcd_interface: for now, hard-code mixed_precision=1

Commit:
invert_clover_eo.c: correct mistaken Qm multiplication from merge

Commit:
invert_eo.c: typo introduced during merge, not sure how it originated

sbacchio committed:
Merge pull request #1 from kostrzewa/merge-MG4QCD

Merge mg4qcd and current master

sbacchio committed:
Moving to DDalphaAMG release

sbacchio committed:
Bug fix: removed conflict with Dsw_psi defined in operator/D_psi.h

sbacchio committed:
Bug fix in MG_check

sbacchio committed:
Added parameters: MGrhoUpdate, MGOMPNumberOfThreads. Renamed paramter: MGdtauSetupIter to MGUpdateSetupIter.

sbacchio committed:
Part of doc. written and setup_mu added

sbacchio committed:
Fixed MGSetupMu

Commit:
read_input: correct verbose output for mg_setup_mu

Commit:
read_spinor: add new flag 'DisableSourceIOChecks' to disable the IO
check for the reading of sources, independently from the gauge IO
checks. 'DisableGaugeIOChecks' parameter added for consistency and
'DisableIOChecks' retained for backwards compatibility. The latter
disables only gauge IO checks, as it used to be.

sbacchio committed:
Merge pull request #2 from kostrzewa/mgsetupmu_readinput

read_input: correct verbose output for mg_setup_mu

Commit:
monitor_forces: output 8 digits for force, otherwise large but steady forces don't fluctuate, making analysis awkward

sbacchio committed:
Merge pull request #4 from Finkenrath/DDalphaAMG

DDalphaAMG

sbacchio committed:
MGMixedPrecision added and some small changes.

sbacchio committed:
merging etmc/master

sbacchio committed:
Merge branch 'etmc-master'

Commit:
remove duplicate g_disable_IO_checks and g_disable_src_IO_checks from read_input.l

sbacchio committed:
Use of MG improved in the heatbath and acceptance step. Some other small chenges

sbacchio committed:
Additional changes for the merge in etmc/tmLQCD

sbacchio committed:
Bug fix.

Commit:
Merge pull request #343 from Finkenrath/master

Merging DDalphaAMG interface to etmc/tmLQCD master

urbach committed:
Merge pull request #338 from kostrzewa/source_io_check

read_spinor: add new flag 'DisableSourceIOChecks'

urbach committed:
Merge pull request #340 from kostrzewa/force_digits

monitor_forces: output 8 digits for force, otherwise large but steady…

sbacchio committed:
- Fixed input reading of MGSetup2KappaMu
- Fixed MG inversion in inlinemeasurment when the configuration is rejected (the new conf wasn't updated in the interface and MGTEST was failing there).

Commit:
Merge pull request #345 from Finkenrath/master

Bug fix in DDalphaAMG interface

sbacchio committed:
Fixed compilation without DDalphaAMG library.

urbach committed:
Merge pull request #347 from Finkenrath/master

Fixing compilation without DDalphaAMG

sbacchio committed:
Changed to %.12f most of the print of kappa, mu, beta, c_sw and related quantities.

martin-ueding committed:
inplementation → implementation

martin-ueding committed:
Insert a comma between two inline equations

martin-ueding committed:
Change the indices $F^{\alpha\beta}$ → $F^{ab}$

The gluon field strength tensor should have a color indices in the
adjoint representation. It also has Lorentz indices. The α and β read
like Dirac indices which I think should not occur on the gluon field
strength tensor.

The Lorentz indices μ and ν are summed over, the left side of the
equation has a, α, b, and β as free indices. Therefore there need to be
some indices a and b on the right side.

martin-ueding committed:
correpsonds → corresponds

martin-ueding committed:
flavor → flavour

martin-ueding committed:
seperately → separately

martin-ueding committed:
Remove double “and”

martin-ueding committed:
preconditioining → preconditioning

Commit:
Merge pull request #352 from Finkenrath/master

Changed to %.12f most of the print of kappa, mu, beta, c_sw and related quantities.

sbacchio committed:
Bug fix: wrong factor for order 3 expansion in correction term for rational approximation.

sbacchio committed:
Merge branch 'master' of github.com:Finkenrath/tmLQCD

Commit:
documentation: adjust parameters and use 'graphicx' package such that the docs can be built using pdflatex, add Makefile for automatic generation of documentation usign latexmk

Commit:
doc: clean target in Makefile should include converted figures

Commit:
Merge pull request #353 from Finkenrath/master

Bug fix in correction term for rational approximation.

sbacchio committed:
Fixed reset of multigrid when several measurements are done with invert or offline_measurments. Changed position MG_reset when configuration is rejected. Fixed update of setup.

Commit:
Merge pull request #356 from Finkenrath/bug_fix

Some bug fix about MG_update.

Commit:
Allocate full tmlQCD spinor as buffer in QPhiX solver interface and
improved output formating.

Commit:
Implemented odd source preparation using QPhiX BLAS and made
the output more verbose.

Commit:
* rearranged order of initialisation of dslash, solver and fermion matrix
objects

* implemented solve of checkerboard 0 (= odd), which converges

* returning the correct number of iterations

Commit:
WIP: Implemented full interface for QPhiX CG with Wilson Fermions.
Does not yet give the same results as tmlQCD.

Commit:
Finished fully working QPhiX Wilson CG Solver interface.

* add additional buffer for QPhiX used in Matrix Multiplication after
solving the normal equation

* Wilson mass calculated correctly

* added correct isign for CG solver (that now solves
M M^dagger y_cb0 = \tilde b_cb0 now first) and additional
multiplication with the fermion matrix (NOT done in QPhiX CG solver)

* corrected the rescaling of the solution vector
x_tmlQCD = 1/(2\kappa) x_qphix

* fixed a bug in the order of arguments passed to convert_lexic_to_eo

plabus commented:
Solver interface works for Wilson fermions using CG.

Commit:
Fixed the solver precision pasted to QPhiX.

QPhiX expects the (non-squared) precision normalised by the norm of the
rhs, so that we have to pass:

RsdTarget = Sqrt( precision_tmlQCD / ||rhs||^2 )

Commit:
Enabled use of BiCGStab solver in QPhiX interface.

The solver is set depending on the solver_flag using a pointer
to AbstractSolver. This however forbids to call the getAypxThreads
routine, b/c it is not defined as a virtual function in AbstractSolver.

Commit:
Getting the number of BLAS threads via reinterpret_cast on the
AbstractSolver pointer.

* BiCGStab has no getters, yet
* check availability of solver now already when creating solver object
* deref pointer of solver object and using operator() directly

Commit:
Using only C++ style casts in invert_eo now.

Commit:
Removed unused variable and comment that does not apply anymore.

Commit:
Changed function name invert_eo to inter_eo_qphix_helper
to prevent collision.

Commit:
Interfaced the creation of Fermion Matrix & Dslash, depending on Fermion
Action chosen, using pointers to base classes.

Commit:
Removed solver tuning (b/c it's not implemented in QPhiX), and set
number of BLAS threads to number of SMT threads.

Commit:
The clover coefficient has to bet set positive.

urbach committed:
Merge pull request #361 from martin-ueding/doc-typos

Fix a couple of typos in the documentation

urbach committed:
Merge pull request #354 from kostrzewa/doc_Makefile_graphicx

documentation: make building easier and allow use of pdflatex

kostrzewa commented:
This is looking good! Would probably prefix the console output with ```qphix_interface: ``` or something like that, just to be explicitly clear where it's coming from.

martin-ueding committed:
Tweak indentation presets

martin-ueding committed:
Add a first draft for the Dslash base classes

martin-ueding committed:
Actually inherit from our new base class

martin-ueding committed:
Add the isign

martin-ueding committed:
Change the method name

martin-ueding committed:
Add forwarding functions for the dslash and achimbdpsi

martin-ueding committed:
Only include MPI if configured

martin-ueding committed:
Add abstract base classes for QPhiX Dslash

martin-ueding committed:
Various fixes

martin-ueding committed:
Use the polymorphic version in the test, does not work yet with TM case

Commit:
qphix_interface work:

1) pass parameters to qphix from input file, new "BeginExternalInverter QPHIX"
   structure implemented for this purpose (warning: no correctness checks here)

2) move _initQphix call into invert_eo_qphix

3) add sample file to call qphix Wilson inverter

4) use angled brackets to include qphix headers

caveats: single precision results in a segfault, only the Wilson operator is currently supported

martin-ueding committed:
Change the mass factor

martin-ueding committed:
Add an application of A after TM Dslash

Commit:
Merge pull request #3 from kostrzewa/juelich_qphix_devel

first attempt to properly pass parameters to qphix from input file

Commit:
Merge pull request #4 from martin-ueding/qphix_devel

Abstract Dslash Interface

martin-ueding committed:
Use tmLQCD copyright header

martin-ueding committed:
Fix the names and factors

Commit:
Merge pull request #5 from martin-ueding/qphix_devel

Fix the A Chi stuff

martin-ueding committed:
Change the license to BSD-3-Clause to make inclusion to QPhiX easier

martin-ueding committed:
Pull out the number of BLAS threads

martin-ueding committed:
Start with clover variants

martin-ueding committed:
Add A_chi and A_inv_chi for clover

martin-ueding committed:
Add a wrapper for four spinors

martin-ueding committed:
Line wrap

martin-ueding committed:
Add a wilson_dslash function

martin-ueding committed:
Add a size() function

martin-ueding committed:
Some documentation

Commit:
check qphix input parameters and set some defaults in the interface

martin-ueding committed:
Typo

Commit:
Merge pull request #7 from kostrzewa/qphix_devel_check_params

check qphix input parameters and set some defaults in the interface

martin-ueding committed:
Revert "Add a wilson_dslash function"

This reverts commit 5ecb2e7ec4d56e6f27b44cfad49664abecf1fb89.

martin-ueding committed:
Merge branch 'master' into qphix_devel

There were strange issues in `config.h.in`. Perhaps not everything was
merged correctly. I tried to just take whatever is defined in `master`
and `qphix_devel` branches, so that everything is properly `#undef`ed.

Conflicts:
	.gitignore
	config.h.in
	configure.in
	invert.c
	invert_clover_eo.c
	invert_eo.c
	read_input.l

martin-ueding committed:
Merge remote-tracking branch 'plabus/qphix_devel' into qphix_devel

martin-ueding committed:
Add some more operators to the wrapper

Commit:
Merge pull request #6 from martin-ueding/qphix_devel

Add Clover

Commit:
Correct A chi, A^-1 chi (has to be \alpha 1I, 1/\alpha 1I)

Commit:
Changed preparation and reconstruction to an action independent scheme.

For this, we need a second Wilson Dslash object for the preparation
and reconstruction part.

martin-ueding committed:
Move get_num_blocks function into tmLQCD interface

Commit:
Merge branch 'qphix_devel' of https://github.com/martin-ueding/tmLQCD into qphix_devel

Commit:
Changed TM Dslash to the correct API

Commit:
Implemented pure twisted-mass case. (INCORRECT RESULTS still)

Commit:
The A chi routine is the ADJOINT of A^{-1} with different factors.

This is why we need an additional minus sign.

Commit:
TM Dslash working without padding only.

Commit:
A chi helper function works with padding now.

Commit:
Set \mu to the correct negative (!) value. Pure TM working now.

Commit:
Corrected the templitisation over the spinor floating point type for QPhiX.

* using FT pointers in reorder routines
* renamed the reorder routines
* corrected const-ness

Commit:
Added sloppy parameter place-holder and changed double precision limit
to 10^-8, which seems to be about the correct boundary for the real
residuum.

Commit:
Clarified comment in reorder_gauge_to_QPhiX.

Commit:
Added const qualifies in reordering routines.

Commit:
qphix_interface: prevent multiple initialisations of QMP topology

Commit:
Merge pull request #8 from kostrzewa/qphix_single_qmp_topo_init

qphix_interface: prevent multiple initialisations of QMP topology

Commit:
hard-revert config.h.in to version from etmc/master and  add TM_USE_QPHIX

Commit:
Merge pull request #9 from kostrzewa/revert_master_config_h_in

align config.h.in in master and qphix_devel
